engines/dradis-api/app/controllers/dradis/ce/api/v1/attachments_controller.rb

```diff
@@ -0,0 +1,83 @@
+module Dradis::CE::API
+  module V1
+    class AttachmentsController < Dradis::CE::API::V1::ProjectScopedController
+      before_action :set_node
+
+      skip_before_action :json_required, :only => [:create]
+
+      def index
+        @attachments = @node.attachments.each(&:close)
+      end
+
+      def show
+        filename = params[:filename]
+        begin
+          @attachment = Attachment.find(filename, conditions: { node_id: @node.id } )
+        rescue
+          raise ActiveRecord::RecordNotFound, "Couldn't find attachment with filename '#{params[:filename]}'"
+        end
+      end
+
+      def create
+        uploaded_files = params.fetch('files', [])
+
+        @attachments = []
+        uploaded_files.each do |uploaded_file|
+          attachment_name = Attachment.available_name(@node, original: uploaded_file.original_filename)
+
+          attachment = Attachment.new(attachment_name, node_id: @node.id)
+          attachment << uploaded_file.read
+          attachment.save
+
+          @attachments << attachment
+        end
+
+        if @attachments.any? && @attachments.count == uploaded_files.count
+          render status: 201
+        else
+          render status: 422
+        end
+      end
+
+      def update
+        filename    = params[:filename]
+        attachment  = Attachment.find(filename, conditions: { node_id: @node.id } )
```

Comment:
I don't think you need to assign the `filename` variable here. You can go straight to `attachment  = Attachment.find(params[:filename], conditions: { node_id: @node.id } )`